src/Native/System.Security.Cryptography.Native/pal_x509ext.cpp

```diff
@@ -3,14 +3,63 @@
 
 #include "pal_x509ext.h"
 
-extern "C" BASIC_CONSTRAINTS* DecodeBasicConstraints(const unsigned char* buf, int32_t len)
+#include <assert.h>
+
+extern "C" X509_EXTENSION* X509ExtensionCreateByObj(ASN1_OBJECT* obj, ASN1_OCTET_STRING* data)
```

Comment:
Please leave the boolean isCritical exposed on the function.  I'll probably need it while implementing the various extension Encode methods.